models/jupiter/solana/jupiter_solana_dca_volume.sql

```diff
@@ -0,0 +1,54 @@
+{{
+  config(
+    schema = 'jupiter_solana',
+    alias = 'dca_volume',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['tx_id', 'outer_instruction_index', 'inner_instruction_index'],
+    post_hook='{{ expose_spells(\'["solana"]\',
+                                      "project",
+                                      "jupiter_solana",
+                                      \'["ilemi"]\') }}')
+}}
+
+SELECT
+    io.call_block_time as block_time
+    , date_trunc('day',io.call_block_time) block_date
+    , io.call_tx_id as tx_id
+    , io.call_outer_instruction_index as outer_instruction_index
+    , COALESCE(io.call_inner_instruction_index,0) as inner_instruction_index
+    , io.account_inputMint as input_mint
+    , COALESCE(tr_in_sol.lamports, tr_in.amount) as in_amount --input/initiate does not contain the swapped amount, we need to grab from the transfer.
+    , tr_in.amount/pow(10,p_in.decimals)*p_in.price as in_amount_usd
+    , io.account_outputMint as output_mint
+    , io.repayAmount as out_amount
+    , io.repayAmount/pow(10,p_out.decimals)*p_out.price as out_amount_usd
+    , dca_t.account_user
+    , io.account_dca
+FROM {{ source('dca_solana','dca_call_fulfillFlashFill') }} io
+LEFT JOIN {{ source('dca_solana','dca_call_transfer') }} dca_t ON dca_t.call_tx_id = io.call_tx_id AND dca_t.call_block_time = io.call_block_time
+LEFT JOIN {{ ref('tokens_solana_transfers') }} tr_in ON tr_in.outer_instruction_index = io.call_outer_instruction_index - 2 --initiate, route, fill, transfer (we want initiate)
```

Comment:
this is a large table, right?
can it be an inner join?
we should apply incremental logic / filters to the join conditions so transfers is cut down to match source read

Reply:
same would apply for any joins below